subprojects/logging/src/main/java/org/gradle/internal/logging/console/DefaultScalableElapsedTimeFormatter.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.logging.console;
+
+public class DefaultScalableElapsedTimeFormatter {
```

Comment:
Looking at the [`Clock` implementation](https://github.com/gradle/gradle/blob/master/subprojects/base-services/src/main/java/org/gradle/internal/time/Clock.java), maybe we could use it directly. Either we use [`prettyTime`](https://github.com/gradle/gradle/blob/master/subprojects/base-services/src/main/java/org/gradle/internal/time/Clock.java#L68-L78) or add another method for shorter pretty time? That made me think should we only display a single time scale unit or should we display something like `3h 54m 2s`?